a1/src/main/java/ro/utcn/spet/a1/repository/memory/InMemoryUserRepository.java

```diff
@@ -0,0 +1,47 @@
+package ro.utcn.spet.a1.repository.memory;
+
+import ro.utcn.spet.a1.model.User;
+import ro.utcn.spet.a1.repository.api.UserRepository;
+
+import java.util.*;
+
+public class InMemoryUserRepository implements UserRepository {
+    private int currentId=1;
+    private final Map<Integer,User> data=new HashMap<Integer,User>();
+
+    @Override
+    public User validateUser(String username, String password) {
```

Comment:
When returning a single object (and not a collection), use Optional instead of null. This prevents NullPointerExceptions. In case of collections, simply return an empty collection.